src/settings.rs

```diff
@@ -122,6 +122,19 @@ impl Settings {
             cfg_path: cfg_path.to_string(),
         })
     }
+
+    pub fn from_server(toml_str: &str) -> Result<Self, ConfigError> {
+        let s = default_config_builder()
+            .add_source(File::from_str(toml_str, config::FileFormat::Toml))
+            .build()?;
+
+        let config: Config = s.try_deserialize()?;
+
+        Ok(Self {
+            config,
+            cfg_path: "from_server".to_string(),
```

Comment:
I think we have room for improvement on this (#866), and I would like to suggest putting an empty string for now. Do you have other suggestions?